modules/gdscript/editor/script_templates/CharacterBody2D/basic_movement.gd

```diff
@@ -22,6 +23,6 @@ func _physics_process(delta: float) -> void:
 	if direction:
 		velocity.x = direction * SPEED
 	else:
-		velocity.x = move_toward(velocity.x, 0, SPEED)
+		velocity.x = move_toward(velocity.x, 0, DECELERATION * delta)
```

Comment:
Let's replace the literal `0` by `0.0` everywhere.

Reply:
Done in a separate commit as it is unrelated to this change. I also replaced the other `0` for the input vector swizzling, I hope that's ok?

Reply:
No need to keep separate commits, you'll have to squash into one at the end anyway

Reply:
Just curious, is there an issue with keeping the 2 independent and functional commits?

Reply:
Because we keep things to one commit when ever possible, to not clutter things and makes managing things easier, what's the issue with having just one commit for one change?

Reply:
> what's the issue with having just one commit for one change?

It's not a single change since it's cosmetics on top of a fix, and I'm touching code that's unrelated to the fix itself (for example the input vector).

Also, It makes reverting or cherry-picking only the fix harder. It is also confusing to see mixed changes along with a fix, as it makes you question if the `0` vs `0.0` was actually part of the problem

Reply:
For these purposes it's a single change, both are part of the same change, if the two need to be separate they should be two PRs instead, the changes aren't going to be reverted separately as the change to float constants is perfectly safe

Reply:
Your call, squashed 🤷